client/state/login/selectors.js

```diff
@@ -76,9 +70,9 @@ export const getTwoFactorPushPollSuccess = ( state ) =>
  * @returns {boolean}        Whether 2FA is enabled
  */
 export const isTwoFactorEnabled = ( state ) => {
-	const twoFactorAuth = get( state, 'login.twoFactorAuth' );
+	const twoFactorAuth = state.login.twoFactorAuth;
 
-	return ! isEmpty( twoFactorAuth );
+	return !! ( twoFactorAuth && Object.keys( twoFactorAuth ).length );
```

Comment:
Seems to me this could be a simple `state.login.twoFactorAuth != null` check. If you look at the reducer, it will never set the state to an empty object: it does the `isEmpty` check itself and sets the state to `null` if the object turns out to be empty.

Reply:
Agreed; good idea!